src/phoenix/server/api/mutations/user_mutations.py

```diff
@@ -40,9 +41,17 @@
 class CreateUserInput:
     email: str
     username: str
-    password: str
+    password: Optional[str] = None
     role: UserRoleInput
     send_welcome_email: Optional[bool] = False
+    auth_method: Optional[AuthMethod] = AuthMethod.LOCAL
+
+    def __post_init__(self) -> None:
+        if self.auth_method is AuthMethod.OAUTH2:
+            if self.password:
+                raise ValueError("Password is not allowed for OAuth2 authentication")
+        elif not self.password:
+            raise ValueError("Password is required for local authentication")
```

Comment:
```suggestion
    def __post_init__(self) -> None:
        if self.auth_method is AuthMethod.OAUTH2:
            if self.password:
                raise BadRequest("Password is not allowed for OAuth2 authentication")
        elif not self.password:
            raise BadRequest("Password is required for local authentication")
```

Given some of the issues that have been coming into community around inscrutable error messages, I want to make sure these APIs make it back to the user.

Reply:
ok sg. thx